engine/table/src/main/java/io/deephaven/engine/table/impl/TimeTable.java

```diff
@@ -114,4 +134,309 @@ protected void destroy() {
         super.destroy();
         UpdateGraphProcessor.DEFAULT.removeSource(this);
     }
+
+    private static class SyntheticDateTimeSource implements
+            ColumnSource<DateTime>,
+            DefaultChunkSource.WithPrev<Values>,
+            MutableColumnSourceGetDefaults.LongBacked<DateTime>,
+            FillUnordered,
+            InMemoryColumnSource {
+
+        private DateTime baseTime;
+        private final long period;
+
+        private SyntheticDateTimeSource(long period) {
+            this.period = period;
+        }
+
+        @Override
+        public DateTime get(long index) {
+            return DateTimeUtils.plus(baseTime, period * index);
```

Comment:
`rowKey` is a better name than `index` these days.
We use this calculation all over the place. Maybe we want a private method for it, or to have everywhere else just call `get`? I'll point out that the default `fillChunk` implementation is just looping on `get`, and that that is what your implementation is doing.